packages/nimble-components/build/validate-multi-color-icons/source/index.js

```diff
@@ -0,0 +1,156 @@
+    __dirname,
+    '../../../src/icons-multicolor'
+);
+
+/**
+ * Count the number of cls-N classes in an SVG string
+ */
+function countLayersInSvg(svgData) {
+    const classMatches = svgData.match(/class="cls-\d+"/g);
+    if (!classMatches) {
+        return 0;
+    }
+    const classNumbers = classMatches.map(match => {
+        const num = match.match(/\d+/);
+        return num ? parseInt(num[0], 10) : 0;
+    });
+    return Math.max(...classNumbers, 0);
+}
+
+/**
+ * Extract SVG data from icon token import
+ */
+function extractSvgFromIconFile(filePath) {
+    const content = fs.readFileSync(filePath, 'utf-8');
+
+    // Match pattern: import { iconName16X16 } from '@ni/nimble-tokens/dist/icons/js';
+    const importMatch = content.match(/import\s*{\s*([a-zA-Z0-9]+)\s*}/);
+    if (!importMatch) {
+        return null;
+    }
+
+    const iconName = importMatch[1];
+
+    // Try to load the icon from nimble-tokens
+    try {
+        // eslint-disable-next-line import/no-dynamic-require, global-require
+        const icons = require('@ni/nimble-tokens/dist/icons/js');
+        const icon = icons[iconName];
+        return icon ? icon.data : null;
+    } catch {
+        return null;
+    }
+}
+
+console.log('[validate-multi-color-icons] Starting validation...\n');
+
+// Validate that nimble-tokens metadata entries match actual files
+console.log(
+    '[validate-multi-color-icons] Validating nimble-tokens metadata matches files...'
+);
+const actualFiles = fs.existsSync(iconsMulticolorDirectory)
+    ? fs
+        .readdirSync(iconsMulticolorDirectory)
+        .filter(f => f.endsWith('.ts'))
+        .map(f => path.basename(f, '.ts')) // Keep in spinal-case: circle-partial-broken
+    : [];
+
+const multiColorIconsSet = new Set(multiColorIcons);
+const missingFromMetadata = actualFiles.filter(
+    name => !multiColorIconsSet.has(name)
+);
+const missingFiles = multiColorIcons.filter(
+    name => !actualFiles.includes(name)
+);
+
+if (missingFromMetadata.length > 0) {
+    console.error(
+        `[validate-multi-color-icons] ERROR: Files exist but not in nimble-tokens metadata: ${missingFromMetadata.join(', ')}`
+    );
+    console.error(
+        '[validate-multi-color-icons] Remediation: Add these icons to multiColorIcons array in nimble-tokens/source/icon-metadata.ts'
+    );
+    process.exit(1);
+}
+
+if (missingFiles.length > 0) {
+    console.error(
+        `[validate-multi-color-icons] ERROR: nimble-tokens metadata lists ${missingFiles.join(', ')} but files don't exist`
+    );
+    console.error(
+        '[validate-multi-color-icons] Remediation: Either create the files in src/icons-multicolor/ or remove from nimble-tokens/source/icon-metadata.ts'
+    );
+    process.exit(1);
+}
+
+console.log(
+    '[validate-multi-color-icons] ✓ nimble-tokens metadata matches files\n'
+);
+
+// Validate layer counts
+console.log('[validate-multi-color-icons] Validating layer counts...');
+let hasErrors = false;
```

Comment:
While this build script is somewhat better since it's no longer relying on log/warning for errors, it's still pretty far from clean code. It uses a mix of error handling mechanisms and it's still reaching out to other packages in strange ways. If the goal is validating contents in nimble-tokens, should this just be a script in that package?